### ISSUE TITLE
Serve docs directly from `rustdoc`command

### ISSUE THREAD

**Issue by hardvain** [steveklabnik/rustdoc#175]
It would be nice to have something like `rustdoc serve`command that generates the documentation and also serves the documentation from a local server.  Auto reload for changes would be a still more nice to have. 

Acceptance Criteria:
* A command that builds the documentation and also serves it from a locally started server
* Watch the files for changes and auto reload the documentation on changes (Extremely nice to have)

**Comment by steveklabnik** [comment]
Yup, this is certainly a thing we've thought about, possibly even as *the* way you look at docs locally.

**Comment by mgattozzi** [comment]
This is similar to #153 but not exactly the same.

**Comment by euclio** [comment]
I'm 👎 on requiring a server to view docs locally, but I guess ember requires it?

**Comment by steveklabnik** [comment]
It does not require it. There are some things that are nicer with one, like pretty URLs, but that's a browser/standards issue, not an Ember thing.

**Comment by TheDan64** [comment]
I'd be interested in giving this a shot! What's the best way to serve files? Would hyper be too low level?

**Comment by steveklabnik** [comment]
So, this is kinda an open question! Given that it's local, I would argue that the smaller, the better.

There are several "simple file servers" in Rust; I even have one of my own: https://github.com/steveklabnik/simple-server/ I'm not saying mine would be the right choice, but I think whatever is simplest is the best option. We don't need to worry about things like scalability when we're talking about browsing docs locally.

**Comment by TheDan64** [comment]
Good point! I'll be sure to check them out. I'm trying out the `notify` crate for watching directories. My initial foray is [here](https://github.com/TheDan64/rustdoc/commit/fe7258b18dd26802b4c913c5148a2dc4e826b9aa).
